src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -229,130 +230,215 @@ public String getUserAnalysisOutputsPage(Model model) {
 	 *
 	 * @param submissionId the ID of the submission
 	 * @param model        Model for the view
+	 * @param principal    Principal {@link User}
 	 * @param locale       User's locale
 	 * @return name of the details page view
 	 */
+
 	@RequestMapping(value = "/{submissionId}", produces = MediaType.TEXT_HTML_VALUE)
-	public String getDetailsPage(@PathVariable Long submissionId, Model model, Locale locale) {
+	public String getDetailsPage(@PathVariable Long submissionId, Model model, final Principal principal,
+			Locale locale) {
 		logger.trace("reading analysis submission " + submissionId);
 		AnalysisSubmission submission = analysisSubmissionService.read(submissionId);
 		model.addAttribute("analysisSubmission", submission);
 
-		boolean canShareToSamples = false;
-		if (submission.getAnalysis() != null) {
-			canShareToSamples = analysisSubmissionSampleProcessor
-					.hasRegisteredAnalysisSampleUpdater(submission.getAnalysis().getAnalysisType());
+		final User currentUser = userService.getUserByUsername(principal.getName());
+
+		if(currentUser != null) {
+			model.addAttribute("isAdmin", currentUser.getSystemRole()
+					.equals(Role.ROLE_ADMIN));
 		}
 
-		model.addAttribute("canShareToSamples", canShareToSamples);
 
+		IridaWorkflow iridaWorkflow = workflowsService.getIridaWorkflowOrUnknown(submission);
 
-		UUID workflowUUID = submission.getWorkflowId();
-		logger.trace("Workflow ID is " + workflowUUID);
+		// Get the name of the workflow
+		AnalysisType analysisType = iridaWorkflow.getWorkflowDescription()
+				.getAnalysisType();
+		model.addAttribute("analysisType", analysisType);
 
+		return "analysis";
+	}
+
+	/**
+	 * Update an analysis email pipeline completion result
+	 *
+	 * @param parameters parameters which include the submission id and
+	 *                   the new email pipeline result value
+	 * @param locale     User's locale
+	 * @return redirect to the analysis page after update
+	 */
+	@RequestMapping(value = "/ajax/update-email-pipeline-result", method = RequestMethod.PATCH)
+	public Map<String, String> ajaxUpdateEmailPipelineResult(@RequestBody AnalysisEmailPipelineResult parameters,
+			Locale locale) {
+		logger.trace("reading analysis submission " + parameters.getAnalysisSubmissionId());
+		AnalysisSubmission submission = analysisSubmissionService.read(parameters.getAnalysisSubmissionId());
+		analysisSubmissionService.updateEmailPipelineResult(submission, parameters.getEmailPipelineResult());
+		logger.trace("Email pipeline result updated for: " + submission);
+
+		String message = "";
+		if (parameters.getEmailPipelineResult()) {
+			message = messageSource.getMessage("analysis.notification.you.will.receive.an.email", new Object[] {},
+					locale);
+		} else {
+			message = messageSource.getMessage("analysis.notification.you.will.not.receive.an.email", new Object[] {},
+					locale);
+		}
+
+		return ImmutableMap.of("result", "success", "message", message);
```

Comment:
Update this to use `HttpServletResponse` and set the status of the response appropriately.  The reason behind this is that your ajax request can catch the error properly instead of use an if statement in the success handler.

Reply:
Fixed in 9c4e640d5c452bf1455f9b93a1b1499a418283e6